src/dotnet/commands/dotnet-compile-native/NativeCompilerCommandApp.cs

```diff
@@ -0,0 +1,147 @@
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Dnx.Runtime.Common.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ProjectModel;
+using NuGet.Frameworks;
+using System.Linq;
+
+// This class is responsible with defining the arguments for the Compile verb.
+// It knows how to interpret them and set default values
+namespace Microsoft.DotNet.Tools.Compiler.Native
+{
+    public delegate bool OnExecute(List<ProjectContext> contexts, NativeCompilerCommandApp compilerCommand);
+
+    public class NativeCompilerCommandApp
+    {
+        private readonly CommandLineApplication _app;
+
+        // options and arguments for compilation
+        private CommandOption _outputOption;
+        private CommandOption _intermediateOutputOption;
+        private CommandOption _frameworkOption;
+        private CommandOption _runtimeOption;
+        private CommandOption _configurationOption;
+        private CommandArgument _projectArgument;
+        private CommandOption _nativeOption;
+        private CommandOption _archOption;
+        private CommandOption _ilcArgsOption;
+        private CommandOption _ilcPathOption;
+        private CommandOption _ilcSdkPathOption;
+        private CommandOption _appDepSdkPathOption;
+        private CommandOption _cppModeOption;
+        private CommandOption _cppCompilerFlagsOption;
+
+        // resolved values for the options and arguments
+        public string ProjectPathValue { get; set; }
+        public string OutputValue { get; set; }
+        public string IntermediateValue { get; set; }
+        public string RuntimeValue{ get; set; }
+        public string ConfigValue { get; set; }
+        public bool IsNativeValue { get; set; }
+        public string ArchValue { get; set; }
+        public string IlcArgsValue { get; set; }
+        public string IlcPathValue { get; set; }
+        public string IlcSdkPathValue { get; set; }
+        public bool IsCppModeValue { get; set; }
+        public string AppDepSdkPathValue { get; set; }
+        public string CppCompilerFlagsValue { get; set; }
+
+        // workaround: CommandLineApplication is internal therefore I cannot make _app protected so baseclasses can add their own params
+        private readonly Dictionary<string, CommandOption> baseClassOptions; 
+
+        public NativeCompilerCommandApp(string name, string fullName, string description)
+        {
+            _app = new CommandLineApplication
+            {
+                Name = name,
+                FullName = fullName,
+                Description = description
+            };
+
+            baseClassOptions = new Dictionary<string, CommandOption>();
+
+            AddCompileParameters();
+        }
+
+        private void AddCompileParameters()
+        {
+            _app.HelpOption("-h|--help");
+
+            _outputOption = _app.Option("-o|--output <OUTPUT_DIR>", "Directory in which to place outputs", CommandOptionType.SingleValue);
```

Comment:
I see the temporary workaround, what is the long term command. "dotnet compile-native  <project.json> | managed dll"? or something like "dotnet compile native" which essentially pipe the commands. I think the "dotnet compile native" pipelining has value since dotnet native can work with any managed compiler ..example "dotnet compile-fsc native".